grakn-engine/src/main/java/ai/grakn/engine/backgroundtasks/taskstatestorage/TaskStateGraphStore.java

```diff
@@ -178,38 +164,27 @@ public Boolean updateState(String id, TaskStatus status, String statusChangeBy,
     }
 
     public TaskState getState(String id) {
-        if(id == null) {
-            return null;
-        }
-
         Optional<TaskState> result = attemptCommitToSystemGraph((graph) -> {
-            Instance instance = graph.getConcept(ConceptId.of(id));
+            Instance instance = graph.getResourcesByValue(id).iterator().next().owner();
```

Comment:
Do you know the resource type? you could make this MUCH faster and scale better by saying `graph.getResourceType("My Type").getResource("Value")`